deepchem/feat/molecule_featurizers/mat_featurizer.py

```diff
@@ -0,0 +1,120 @@
+from deepchem.feat.base_classes import MolecularFeaturizer
+from deepchem.utils.molecule_feature_utils import one_hot_encode
+import numpy as np
+from rdkit import Chem
+from sklearn.metrics import pairwise_distances
+
+
+class MATFeaturizer(MolecularFeaturizer):
+  """
+  This class is a featurizer for the Molecule Attention Transformer [1]_.
+  The featurizer accepts an RDKit Molecule, and 2 booleans (add_dummy_node and one_hot_formal_charge) as arguments.
+  The returned value is a tuple which consists of molecular graph descriptions:
+    - Node Features
+    - Adjacency Matrix
+    - Distance Matrix
+  """
+
+  def __init__(
+      self,
+      mol: Chem.rdchem.Mol,
+      add_dummy_node: bool = True,
+      one_hot_formal_charge: bool = True,
+  ):
+    """
+    Parameters
+    ----------
+    mol (rdchem.Mol): rdkit.Chem.rdchem.Mol
+      RDKit Mol object.
+    add_dummy_node: bool, default True
+      If True, a dummy node will be added to the molecular graph.
+    one_hot_formal_charge: bool, default True
+      If True, formal charges on atoms are one-hot encoded.
+
+    References
```

Comment:
Can you move the reference section to the main docstring for the class?

Reply:
Done!